app/api/calendars.js

```diff
@@ -0,0 +1,159 @@
+      const toRead = await models.Calendar.findByPk(req.params.id)
+
+      //verify valid id
+      if (!toRead) {
+        return res.status(400).json(error.parse('calendar-400', idNotFoundDevMessage()))
+      }
+
+      //checar visibilidade do processo
+      const user = await findUserByToken(req.headers['x-access-token'], app.get('jwt_secret'), models)
+      const visibleCallId = await filterVisibleByCallId(toRead.call_id, user, models)
+      if (!visibleCallId) {
+        return res.status(400).json(error.parse('calendar-400', idNotFoundDevMessage()))
+      }
+
+      //return result
+      return res.json(toRead)
+
+      //if error
+    } catch (err) {
+      return res.status(500).json(error.parse('calendar-500', unknownDevMessage(err)))
+    }
+  }
+
+  //Calendar update
+  api.update = async (req, res) => {
+    try {
+      const toUpdate = await models.Calendar.findByPk(req.params.id)
+
+      //verify valid id
+      if (!toUpdate) {
+        return res.status(400).json(error.parse('calendar-400', idNotFoundDevMessage()))
+      }
+
+      //validation
+      const validationErrors = await validateBody(req.body, models, 'update', toUpdate)
+      if (validationErrors) {
+        return res.status(400).json(error.parse('calendar-400', validationDevMessage(validationErrors)))
+      }
+
+      //permission
+      const permissionErrors = await validatePermission(req, models, toUpdate)
+      if (permissionErrors) {
+        return res.status(401).json(error.parse('calendar-401', unauthorizedDevMessage(permissionErrors)))
+      }
+
+      //try to create
+      const updated = await toUpdate.update(req.body)
+      await updated.reload() //para que o retorno seja igual ao de api.read.
+      return res.status(201).json(updated)
+
+      //if error
+    } catch (err) {
+      console.log('Erro:', err)
+      return res.status(500).json(error.parse('calendar-500', unknownDevMessage(err)))
+    }
+  }
+
+  //Calendar delete
+  api.delete = async (req, res) => {
+    try {
+      const toDelete = await models.Calendar.findByPk(req.params.id)
+
+      //verify valid id
+      if (!toDelete) {
+        return res.status(400).json(error.parse('calendar-400', idNotFoundDevMessage()))
+      }
+
+      //permission
+      const permissionErrors = await validatePermission(req, models, toDelete)
+      if (permissionErrors) {
+        return res.status(401).json(error.parse('calendar-401', unauthorizedDevMessage(permissionErrors)))
+      }
+
+      //try to delete
+      await models.Calendar.destroy({
+        where: { id: req.params.id },
+        individualHooks: true
+      }).then(_ => res.sendStatus(204))
+
+      //if error
+    } catch (err) {
+      if (err.name === 'ForbbidenDeletionError')
+        return res.status(403).json(error.parse('calendar-403', forbbidenDeletionDevMessage(err)))
+      return res.status(500).json(error.parse('calendar-500', unknownDevMessage(err)))
+    }
+  }
+
+  api.list = async (req, res) => {
```

Comment:
Lança erro quando envio um id só (calendar-500, erro interno do servidor). Com a mensagem "callIds.map is not a function". 

Fiz a chamada com a seguinte URL:
http://localhost:3000/v1/calendars/?call_ids=cb271efe-c7e0-4c10-9e86-69443a484e78

Porém o servidor respondeu quando eu dupliquei o mesmo valor 
http://localhost:3000/v1/calendars/?call_ids=cb271efe-c7e0-4c10-9e86-69443a484e78&call_ids=cb271efe-c7e0-4c10-9e86-69443a484e78

Se não me engano em SelectiveProcesses a gente enviava vários ids separados por vírgula. Temos que decidir se usamos esse padrão ou o outro, pra deixar um só.

Reply:
Não sei por que o segundo deu certo, mas...

O formato no qual vc está enviando não é o formato no qual eu codifiquei. Quando se quer enviar um vetor pela query string, existe uma forma mais pratica que é assim:

http://localhost:3000/v1/calendars/?call_ids[]=cb271efe-c7e0-4c10-9e86-69443a484e78

ou mais de um:

http://localhost:3000/v1/calendars/?call_ids[]=aaa&call_ids[]=bbb

Dessa forma não é necessário 'traduzir' a string quando ela chega no backend.